### PR TITLE
refactor: unify multi-seed-keychain and keychain

### PR DESCRIPTION
**Pull request by mvayngrib**
depends on #50 , per @sparten11740 's suggestion to unify the two

### CODE REVIEW

**Review by diegomura — APPROVED** [review]
utACK

Agree on unifying. Should this be breaking?

File: features/keychain/module/keychain.js [hunk 1]
```diff
@@ -43,31 +44,41 @@ export class Keychain extends ExodusModule {
...
-  #getPrivateHDKey = (keyId) => {
+  #getPrivateHDKey = ({ seedId, keyId }) => {
     throwIfInvalidKeyIdentifier(keyId)
-    throwIfInvalidMasters(this.#masters)
+
+    if (typeof seedId !== 'string' || !seedId) {
+      throw new Error('seedId must be a BIP32 key identifier in hex encoding')
+    }
+
+    if (!this.#masters[seedId]) throw new Error(`seed "${seedId}" is not initialized`)
```
Comment (sparten11740) [inline features/keychain/module/keychain.js:72]: nit
```suggestion
    assert(typeof seedId === 'string', 'seedId must be a BIP32 key identifier in hex encoding') 
    assert(this.#masters[seedId], `seed "${seedId}" is not initialized`)
```

**Review by sparten11740 — APPROVED** [review]
thanks, I like this much better

utACK

**Comment by mvayngrib** [comment]
@diegomura the parent is breaking, i was planning to merge it into the parent and merge from there

**Pull request merged**